### ISSUE TITLE
Quality of search results

### ISSUE THREAD

**Issue by impredicative** [pypa/warehouse#5257]
**What's the problem this feature will solve?**
The quality of search results for packages on pypi leaves a lot to be desired. I found the older pypi to have a far better quality of results.

**Describe the solution you'd like**
Use a different search mechanism. How does Google show substantially better quality search results?

**Comment by ewdurbin** [comment]
Hi @impredicative! PyPI's search functionality could indeed use improvement.

As a community supported project, prioritizing features and improvements to PyPI comes down to two things.

1) Community contributions. All users of PyPI are encouraged to discuss and implement improvements to the project! No matter the size or focus of the contribution, the maintainers are always happy to review, and occasionally are able to assist in getting things ready for merge.

2) Funding. Ultimately some features and improvements are too large for us to rely on volunteers. We regularly seek funding for specific projects, but historically things like security, accessibility, and localization have been higher priority for sponsors/donors. If you know of a team for whom funding search improvements would make an impact, please introduce us!

> How does Google show substantially better quality search results?

In short, they are a multibillion dollar company who specializes in search.

As far as this specific issue is concerned, there isn't anything directly actionable.

**Comment by impredicative** [comment]
Well, how did the old pypi show better search results (in my opinion)? What technique did it use anyway?

**Comment by ewdurbin** [comment]
The codebase for the original implementation of PyPI is viewable at https://github.com/pypa/pypi-legacy.

Effectively the search implementation was [this database query](https://github.com/pypa/pypi-legacy/blob/a418131180a14768328713df5533c600c425760d/store.py#L613-L661), which was eventually [emulated using elastic search](https://github.com/pypa/pypi-legacy/blob/a418131180a14768328713df5533c600c425760d/store.py#L663-L719).

Most likely reason for the dramatic change in results with this codebase is the inclusion of project descriptions in the search corpus.
